e2e/testcases/multi_sync_test.go

```diff
@@ -415,10 +415,27 @@ func TestConflictingDefinitions_RootToRoot(t *testing.T) {
 	nt.RootRepos[rootSync2].Add(podRoleFilePath, rootPodRole())
 	nt.RootRepos[rootSync2].CommitAndPush("add conflicting pod owner role")
 
-	nt.T.Logf("The admission webhook should deny the update request in Root %s", rootSync2)
-	nt.WaitForRootSyncSyncError(rootSync2, applier.ApplierErrorCode, "denied the request")
-	nt.T.Logf("Root %s should also get surfaced with the conflict error", configsync.RootSyncName)
-	nt.WaitForRootSyncSyncError(configsync.RootSyncName, status.ManagementConflictErrorCode, "declared in another repository")
+	// When the webhook is enabled, it will stop adoption of managed objects.
+	// Only the second RootSync will show the conflict error from the applier,
+	// But the first RootSync will also be updated by the second RootSync's reconciler.
+	nt.T.Logf("RootSync %s should show a conflict error because the webhook denied the update request", rootSync2)
+	nt.T.Logf("RootSync %s should show a conflict error because that the other reconciler cross-posted the error", configsync.RootSyncName)
+	// Run the Watches in parallel for speed and consistency.
+	waitInParallel(
+		func() {
+			nomostest.WatchForObject(nt, kinds.RootSyncV1Beta1(), configsync.RootSyncName, configsync.ControllerNamespace,
+				[]nomostest.Predicate{
+					nomostest.RootSyncHasSyncError(nt, status.ManagementConflictErrorCode, "declared in another repository"),
+				})
+		},
+		func() {
+			nomostest.WatchForObject(nt, kinds.RootSyncV1Beta1(), rootSync2, configsync.ControllerNamespace,
+				[]nomostest.Predicate{
+					nomostest.RootSyncHasSyncError(nt, status.ManagementConflictErrorCode, "declared in another repository"),
```

Comment:
For the first RootSync, it should show the `denied the request` error.

Reply:
AFAICT, "denied the request" doesn't exist in the code. 

"declared in another repository" is from the `KptManagementConflictError` which is returned by both the applier and the remediator.

Reply:
`denied the request` is from the admission webhook: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/errors/statuserror.go#L29.

The error will first show up because the admission webhook denied the request. The `declared in another repository` error is surfaced by the conflicting remediator. The RootSync status will flip.

Technically, we should expect the `denied the request` error for the first RootSync, especially the cross-RootSync update is going to be removed.

Reply:
Thanks for finding the `denied the request` code. Not sure why it didn't come up in my search.

Should we expect both the `denied the request` and the `declared in another repository` then? The tests are passing as-is, which seems to indicate that the conflict error from the remediator is definitely being propagated from rootSync2 to rootSync1. Adding `denied the request` would confirm that it also was detected via the apply being rejected by the webhook. But we probably wouldn't expect to see them at the same time...

Reply:
Yes, we can expect both errors. Both RootSync1's applier and RootSync2's remediator updates the RootSync1's status. The RootSync1's status was expected to bounce back and forth to report the conflict in different ways. For this test, we care more about the `denied the request` error.

As we're planning to remove the cross-RootSync update, we can validate the `denied the request` error for now.

Reply:
I've updated the test to check for all the possible errors.

When we disable cross-RootSync update, some of these will fail and we can remove them.